orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/model/DeploymentModelSpec.groovy

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model
+
+import com.fasterxml.jackson.annotation.JsonBackReference
+import com.fasterxml.jackson.annotation.JsonManagedReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import groovy.transform.Canonical
+import org.yaml.snakeyaml.Yaml
+import spock.lang.Specification
+
+/**
+ * DeploymentModelSpec.
+ */
+class DeploymentModelSpec extends Specification {
+/* jackson foo:
```

Comment:
ah, ignore this jackon bit, was looking at options for jackson mapping to address the case of modelling things as map<`id`, `the thing`> and `the thing` being able to know its id